app/project/author/controllers.py

```diff
@@ -0,0 +1,72 @@
+from flask_restful import Resource
+
+from project.db import db, Author, Book
+from project.utils.auth import require_admin
+from .parsers import AuthorAddParser, AuthorParser
+
+class List(Resource):
+
+    def get(self, user_data):
+        return [a.json for a in Author.query.all()]
+
+class Add(Resource):
+
+    @require_admin
+    def post(self, user_data):
+        parser = AuthorAddParser(bundle_errors=True)
+        args = parser.parse_args()
+
+        author = Author()
+        author.name = args['name']
+        author.surname = args['surname']
+        author.www = args['www']
+        author.slug = "{0}-{1}".format(args['name'].lower(), args['surname'].lower())
+
+        test = Author.query.filter_by(slug=author.slug).first()
```

Comment:
more pythonic approach is to just try to add author to the session and if fail on `commit` catch `IntegrityError` and return error.